src/main/java/ch/heigvd/pro/b04/moderators/Moderator.java

```diff
@@ -89,4 +94,47 @@ public ServerPoll newPoll(ServerPollRepository repository, ClientPoll poll) {
         .title(poll.getTitle())
         .build());
   }
+
+  /** Verifies that a given moderator has access to a poll.
+   *
+   * @param serverPollRepository The serverpoll repository
+   * @param idPoll The id of the poll
+   * @return The poll associated with idPoll if it belongs to the moderator
+   * @throws PollNotExistingException is thrown if the poll doesn't belong to the moderator or
+   *         doesn't exist
+   */
+  public ServerPoll getPollWithId(ServerPollRepository serverPollRepository, Integer idPoll)
+      throws PollNotExistingException {
+
+    List<ServerPoll> pollList = serverPollRepository.findByModeratorAndId(this, idPoll);
+    if (pollList.isEmpty()) {
+      throw new PollNotExistingException();
+    } else if (pollList.size() > 1) {
+      throw new IllegalPollStateException();
```

Comment:
Unchecked `IllegalPollStateException` (more info on the comment of the class itself).

Reply:
This Exception is deleted by refactoring of `findByModeratorAndId`